Design/#46  로그인 / 알림 / 마이페이지 화면 Hi-fi 반영하기

devPaidion opened the pull request:
## ✨ PR 유형

어떤 변경 사항이 있나요??
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] 사용자 UI 디자인 변경 및 추가
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제


## 작업 이슈 번호를 입력해주세요
- Closes #46 


## 🛠️ 작업내용

#### 로그인
- 로그인 화면 Hi-fi 반영
- 최초 가입시 이름 설정 화면 Hi-fi 반영
- 최초 가입시 이름 설정 화면 글자 제한 및 햅틱 이벤트 추가
- 최초 가입시 이름 설정 토스트 메시지 추가

#### 알림
- 알림 화면 Hi-fi 반영
- 알림 탭바 수신함에 팀스페이스별 안읽은 알림 숫자 처리

#### 마이페이지
- 마이페이지 화면 Hi-fi 반영
- 마이페이지 이름 변경 화면 글자 제한 및 햅틱 이벤트 추가
- 마이페이지 이름 변경 화면 토스트 메시지 추가
- 마이페이지 이름 변경 화면 사항 저장 관련 모달 추가
- 마이페이지 로그아웃 모달 추가
- 마이페이지 회원탈퇴 모달 추가


## ✅ 체크 포인트
- 전반적인 Hi-fi 반영
- 마이페이지 이름 수정 화면 변경사항 미저장 경고
- 마이페이지 이름 글자 제한 토스트 메시지
- 로그아웃 및 회원탈퇴 기능 정상 여부
- 알림 뱃지 카운팅 기능 정상 여부
- 로딩스피너 기능 정상 여부


## 📋 시연 영상

### 로그인
https://github.com/user-attachments/assets/51de0c1f-e04f-4f0b-84cc-886a2145c758

### 알림 화면
https://github.com/user-attachments/assets/18e81cde-2b22-42f6-9159-adcdb924e7b3

### 마이페이지
https://github.com/user-attachments/assets/992ac435-5124-43dd-a909-00b006c0a80f

### 로딩스피너
https://github.com/user-attachments/assets/70f884c8-9580-496c-9bb8-b4c0f34b987a

devPaidion committed:
💄[Design]#46 - 앱로고 에셋 추가

- 로그인 화면에서 보여주는 용도로 추가

devPaidion committed:
💄[Design]#46 - 로그인 화면 1차 Hi-fi 적용

Velko716 committed:
✨[Feat]#46 - clearGlass 리퀴드 글래스 버튼 분기처리 구현

devPaidion committed:
✨[Feat]#46 - 애플 로고 추가

devPaidion committed:
✨[Feat]#46 - 애플 로고 추가

devPaidion committed:
✨[Feat]#46 - 애플 로그인 커스텀 버튼 추가

devPaidion committed:
✨[Feat]#46 - 애플 로그인 커스텀 버튼 추가

devPaidion committed:
💄[Design]#46 - 로그인 화면 애플 커스텀 버튼 적용

devPaidion committed:
💄[Design]#46 - 로그인 화면 애플 커스텀 버튼 적용

devPaidion committed:
🔧[Chore]#46 - 로그인 ProgressView 스타일 수정

- 로딩중일 때 잘 보이도록 흰색으로 색상 수정

devPaidion committed:
🔧[Chore]#46 - 로그인 ProgressView 스타일 수정

- 로딩중일 때 잘 보이도록 흰색으로 색상 수정

devPaidion committed:
💄[Design]#46 - ActionButton Hi-fi 반영

# Conflicts:
#	DanceMachine/DanceMachine/Common/Components/ActionButton.swift

devPaidion committed:
💄[Design]#46 - ActionButton Hi-fi 반영

devPaidion committed:
✨[Feat]#46 - 햅틱 매니저 추가

- 햅틱 이벤트를 앱 전역에서 사용 가능
- 저전력 모드에서는 사용 불가

devPaidion committed:
✨[Feat]#46 - 햅틱 매니저 추가

- 햅틱 이벤트를 앱 전역에서 사용 가능
- 저전력 모드에서는 사용 불가

devPaidion committed:
💄[Design]#46 - 이름 설정 화면 Hi-fi 반영

devPaidion committed:
💄[Design]#46 - 이름 설정 화면 Hi-fi 반영

devPaidion committed:
🔧[Chore]#46 - 최초 로그인시 이름 설정 화면 첫 글자 공백 금지 기능 추가

devPaidion committed:
🔧[Chore]#46 - 최초 로그인시 이름 설정 화면 첫 글자 공백 금지 기능 추가

devPaidion committed:
✨[Feat]#46 - 알림목록 컴포넌트 추가

- 수신함의 알림목록을 컴포넌트로 분리

devPaidion committed:
✨[Feat]#46 - 알림목록 컴포넌트 추가

- 수신함의 알림목록을 컴포넌트로 분리

devPaidion committed:
💄[Design]#46 - ToolbarCenterTitle Hi-fi 반영

Velko716 committed:
✨[Feat]#42 - 토스트 메세지 공통 컴포넌트 제작

devPaidion committed:
✨[Feat]#46 - 최초 가입시 이름 설정 화면 토스트 메시지 적용

devPaidion committed:
✨[Feat]#46 - 최초 가입시 이름 설정 화면 토스트 메시지 적용

devPaidion committed:
🔧[Chore]#46 - is_read 필드명 하드 코딩 수정

devPaidion committed:
🔧[Chore]#46 - is_read 필드명 하드 코딩 수정

devPaidion committed:
🔥[Del]#46 - user_notification 서브 콜렉션의 asDictionary 삭제

- 불피요한 extension 프로퍼티 삭제

devPaidion committed:
🔥[Del]#46 - user_notification 서브 콜렉션의 asDictionary 삭제

- 불피요한 extension 프로퍼티 삭제

devPaidion committed:
✨[Feat]#46 - user_notification 콜렉션의 created_at 필드 프로퍼티 추가

- 서버에 쿼리를 보낼 때 필요해서 추가함

devPaidion committed:
✨[Feat]#46 - user_notification 콜렉션의 created_at 필드 프로퍼티 추가

- 서버에 쿼리를 보낼 때 필요해서 추가함

devPaidion committed:
✨[Feat]#46 - 팀스페이스별 안읽은 알림 개수를 가져오는 메서드 추가

- 팀스페이스별로 안읽은 알림 개수 가져오는 메서드 추가
- UI에 보여줄 프로퍼티 추가

devPaidion committed:
✨[Feat]#46 - 팀스페이스별 안읽은 알림 개수를 가져오는 메서드 추가

- 팀스페이스별로 안읽은 알림 개수 가져오는 메서드 추가
- UI에 보여줄 프로퍼티 추가

devPaidion committed:
✨[Feat]#46 - 탭바 수신함 알림 개수 뱃지 카운팅 추가

- 들여쓰기 수정
- NotificationManager 싱글톤에서 관리하는 팀스페이스별 안읽은 알림 개수 뱃지에 반영

devPaidion committed:
✨[Feat]#46 - 홈화면에서 팀스페이스 바뀌면 수신함에 안읽음 뱃지 갱신

devPaidion committed:
✨[Feat]#46 - 알림 누르면 탭바 수신함 뱃지 카운팅 하나 줄이기

devPaidion committed:
✨[Feat]#46 - 알림 누르면 탭바 수신함 뱃지 카운팅 하나 줄이기

devPaidion committed:
🔧[Chore]#46 - 알림화면 뷰모델에서 userId 하드코딩 삭제

- InboxViewModel 에서 하드코딩 된 userId 삭제

devPaidion committed:
🔧[Chore]#46 - 알림화면 뷰모델에서 userId 하드코딩 삭제

- InboxViewModel 에서 하드코딩 된 userId 삭제

devPaidion committed:
💄[Design]#46 - 마이페이지 컴포넌트 Hi-fi 반영

devPaidion committed:
💄[Design]#46 - 마이페이지 컴포넌트 Hi-fi 반영

devPaidion committed:
🔧[Chore]#46 - ToastView 카단 코드 적용

devPaidion committed:
🔧[Chore]#46 - ToastView 카단 코드 적용

devPaidion committed:
💄[Design]#46 - 구분선 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 구분선 Hi-fi 적용

devPaidion committed:
✨[Feat]#46 - 변경사항 미저장 알럿 카단 코드 적용

devPaidion committed:
✨[Feat]#46 - 변경사항 미저장 알럿 카단 코드 적용

devPaidion committed:
🔧[Chore]#46 - 최초 로그인시 이름 설정 화면 토스트 메시지 아이콘 설정

devPaidion committed:
🔧[Chore]#46 - 최초 로그인시 이름 설정 화면 토스트 메시지 아이콘 설정

devPaidion committed:
🔧[Chore]#46 - 뒤로가기 버튼에 커스텀 기능 추가

- 뒤로가기 눌렀을 때 원하는 기능 실행될 수 있도록 하는 기능 추가
- 예를 들어, 변경사항 미적용시 경고창 화면에 보이게 할 수 있음

devPaidion committed:
🔧[Chore]#46 - 뒤로가기 버튼에 커스텀 기능 추가

- 뒤로가기 눌렀을 때 원하는 기능 실행될 수 있도록 하는 기능 추가
- 예를 들어, 변경사항 미적용시 경고창 화면에 보이게 할 수 있음

devPaidion committed:
💄[Design]#46 - 이름 설정 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 이름 설정 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 계정 설정 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 계정 설정 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 이용 약관 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 이용 약관 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 개인정보처리방침 화면 Hi-fi 적용

devPaidion committed:
💄[Design]#46 - 개인정보처리방침 화면 Hi-fi 적용

devPaidion committed:
✨[Feat]#46 - DirAct를 만든 사람들 화면 추가

devPaidion committed:
✨[Feat]#46 - DirAct를 만든 사람들 화면 추가

devPaidion committed:
✨[Feat]#46 - 마이페이지 앱 버전 및 DirAct를 만든 사람들 화면 추가

devPaidion committed:
✨[Feat]#46 - 마이페이지 앱 버전 및 DirAct를 만든 사람들 화면 추가

devPaidion committed:
🔧[Chore]#46 - RootView tint 원상복귀

devPaidion committed:
🔧[Chore]#46 - RootView tint 원상복귀

devPaidion committed:
Merge branch 'design/#46' of https://github.com/DeveloperAcademy-POSTECH/2025-C6-A12-VelKaJeJulLiPa into design/#46

# Conflicts:
#	DanceMachine/DanceMachine/Common/Components/Toolbar/ToolbarCenterTitle.swift
#	DanceMachine/DanceMachine/Common/Components/Toolbar/ToolbarLeadingBackButton.swift
#	DanceMachine/DanceMachine/Navigation/MainNavigationRoutingView.swift
#	DanceMachine/DanceMachine/Navigation/NavigationDestination.swift
#	DanceMachine/DanceMachine/Presentations/Home/Views/HomeView/HomeView.swift
#	DanceMachine/DanceMachine/Presentations/MyPage/Views/AccountSettingView.swift
#	DanceMachine/DanceMachine/Presentations/MyPage/Views/EditNameView.swift
#	DanceMachine/DanceMachine/Presentations/MyPage/Views/MyPageView.swift
#	DanceMachine/DanceMachine/Presentations/RootView.swift

devPaidion committed:
✨[Chore]#46 - MainNavigationRoutingView 리베이스 꼬임 수정

devPaidion committed:
🔧[Chore]#46 - 토스트 메시지 아이콘 넣기

devPaidion committed:
🔧[Chore]#46 - 리베이스에서 꼬인 수신함 뱃지 카운팅 반영

devPaidion committed:
🔧[Chore]#46 - 로그아웃 및 회원탈퇴 이후 탭라우터 홈으로 변경

- 로그아웃 및 회원탈퇴 이후 탭라우터 홈으로 변경하고 기존 탭라우터 destination 모두 삭제

devPaidion committed:
🔧[Chore]#46 - 계정 설정 화면 중괄호 추가

devPaidion committed:
🔥[Del]#46 - RootView 리베이스 꼬임 코드 삭제

devPaidion committed:
🔧[Chore]#46 - 수신함 뱃지 카운팅 팀스페이스별 갱신 로직 추가

- 리베이스에서 꼬인 코드 복구

devPaidion committed:
🔧[Chore]#46 - 푸시 알림 로그아웃 한 사용자한테 가지 않게 하기

- 로그아웃 하면 FCM 토큰이 빈 문자열로 저장되고 이를 푸시 알림 보내는 시점에 감지해서 푸시 알림을 보내지 않는 기능 추가

devPaidion committed:
🔧[Chore]#46 - 로그아웃 할 때 푸시 알림 빈 문자열로 저장

devPaidion committed:
🔧[Chore]#46 - FCM 토큰 발급받으면 UserDefaults에 저장

- 로그인 할 때 로컬에 저장했던 FCM토큰을 서버에 저장

devPaidion committed:
🔧[Chore]#46 - 수동으로 로그인할 때, UserDefaults로 저장된 FCM토큰 저장

- 최초 로그인 할 때와 다시 로그인 할 때 모두 UserDefaults로 저장된 FCM토큰을 활용하여 서버에 저장합니다.

devPaidion committed:
🐛[Fix]#46 - 수동 로그인 사용자 데이터 저장 및 불러오기 순서 변경

- 마지막 로그인 시각 업데이트 -> FCM 토큰 저장 -> DB에 저장된 사용자 정보 불러와서 세팅

devPaidion committed:
🔧[Chore] Cloud Functions 로그 남기기 수정

Velko716 commented on DanceMachine/DanceMachine/Common/Components/MyPage/MyPageNavigationRow.swift:
```diff
@@ -8,36 +8,40 @@
+          .font(.headline2SemiBold)
+          .foregroundStyle(.labelNormal)
+      }
+      .padding()
```
.padding() 디폴드 값이 15로 알고 있는데, 이 부분 피그마에 15로 간격설정이 혹시 되어있나여...?

devPaidion replied:
저는 아래 스크린샷처럼 `16` 으로 되어 있는 것 같아요! 더불어 전반적으로 제가 Hi-fi 구분선을 넣지 않아서 관련해서 모두 수정해서 올려두었습니다!

<img width="1920" height="1243" alt="Screenshot 2025-11-08 at 9 41 59 AM" src="https://github.com/user-attachments/assets/281f26f3-7b3f-4e1f-9aaa-b9f8336bb682" />

Velko716 replied:
.padding()이 디폴트값으로 15간격을 줍니다..! 그래서 .padding(.vertical, 16) 이렇게 명시를 해줘야 정확하게 16의 간격을 줘요!

devPaidion replied:
넵! 그렇게 코드 구현해서 올려두었어요 :)

Velko716 replied:
엇.. 근데 전체 .padding()이 필요한 부분이 있나요..?

Velko716 commented on DanceMachine/DanceMachine/Common/Components/MyPage/MyPageRow.swift:
```diff
@@ -8,26 +8,29 @@
+        .font(.headline2Medium)
+        .foregroundStyle(.labelNormal)
     }
+    .padding()
```
.padding() 디폴드 값이 15로 알고 있는데, 이 부분 피그마에 15로 간격설정이 혹시 되어있나여...?

devPaidion replied:
이전 대화와 동일한 내용으로 답변 드렸습니다!

Velko716 commented on DanceMachine/DanceMachine/Common/Components/ToastView.swift:
```diff
@@ -33,6 +35,25 @@ struct ToastView: View {
   }
 }
 
+enum ToastIcon: String {
```
이 부분은 파일 생성해서 따로 파도 좋을 것 같아요!

devPaidion replied:
Enum 폴더에 따로 분리하도록 해두었습니다!

Velko716 commented on DanceMachine/DanceMachine/Common/Components/Login/CustomAppleSignInButton.swift:
```diff
@@ -0,0 +1,41 @@
+
+import SwiftUI
+
+// FIXME: - Hi-fi 적용하기
```
애플 로그인 버튼 커스텀 하신 이유가 무엇일까요??
https://developer.apple.com/kr/design/human-interface-guidelines/sign-in-with-apple
HIG보시면 정말 사이즈랑 크기를 정확하게 맞춰야 심사에 통과하기 때문에, 굳이 쓸 이유가 없다면 제공해주는 버튼을 사용하는게 유리하긴 합니다...

devPaidion replied:
기본 컴포넌트는 한국어가 불가해서 커스텀해야할 것 같아요! 자동으로 언어에 맞춰서 레이블이 변경되지는 않더라고요!
`Apple로 로그인`이라는 레이블 자체는 문제가 없을 것 같습니다만... 말씀하신대로 스타일(글자 크기, 이미지 크기, 비율 등)은 한번 HIG에서 안내한대로 디자인되고 구현되었는지 확인해봐야할 것 같습니다! 오늘 중으로 디자이너분과 애기해서 진행하고 추후 아래에 업데이트 하겠습니다!

devPaidion replied:
제이콥과 이야기해서 HIG 안내사항 대로 디자인 스펙 확인했습니다! 코드 상으로는 버튼 높이를 `54`로 고정, 시스템 폰트 사용 그리고 폰트 크기 `19`, weight는 `semiBold` 사용하는 것으로 수정했습니다! 아무래도 큰 문제는 없을 것 같아요 🥺

Velko716 replied:
기본 컴포넌트도 한글 커스텀 됩니다.
<img width="728" height="567" alt="스크린샷 2025-11-08 오후 2 49 44" src="https://github.com/user-attachments/assets/0961e3f6-f6bd-4fa5-9212-7c3c11edb3fe" />

이미지를 참고해서 보시면 원래 english가 디폴트 값으로 설정되어 있는데 한국어로 변경해주시면 돼요. (추가적으로, 이 부분은 제가 PaperKit구현하면서 한국어를 디폴트값으로 설정했습니다. 따로 설정하지 않으셔도 돼요.)

그리고 HIG 대로 구현하셨다고 하더라도, 정말 하나의 오차라도 발생하면 이 부분은 바로 리젝이라서 왠만하면... 기본 컴포넌트 사용하는걸 추천드리고 싶습니다...

Velko716 reviewed (commented):
고생하셨습니다.

devPaidion committed:
🔧[Chore]#46 - 토스트 아이콘 Enum 파일 분리

devPaidion committed:
🔥[Del]#46 - ToastIcon Enum 코드 파일 분리에 따른 코드 삭제

devPaidion committed:
🔧[Chore]#46 - 로그인시 사용자 정보 관련 서버 호출 순서 변경

- 가장 최신 사용자 정보를 서버에서 받아와서 클라이언트에서 사용할 수 있도록 수정

devPaidion committed:
💄[Design]#46 - 애플 로그인 버튼 HIG 디자인 반영

- 시스템 폰트, 글자 크기 19, 버튼 높이 54로 고정

devPaidion committed:
💄[Design]#46 - 마이페이지 구분선 적용

devPaidion committed:
🔧[Chore]#46 - appIcon 을 AppIcon으로 이름 수정

devPaidion committed:
✨[Feat]#46 - 임시 미모지 Asset 추가

devPaidion committed:
🔥[Del]#46 - 애플 로고 이미지 삭제 (내장된 시스템이미지로 불러오기)

devPaidion committed:
🔧[Chore]#46 - "AppLogo"로 앱로고 이미지 불러오기

devPaidion committed:
✨[Feat]#46 - TeamMember Enum 추가

devPaidion committed:
✨[Feat]#46 - 팀멤버 정보 컴포넌트 추가

devPaidion committed:
🔧[Chore]#46 - DirAct를 만든 사람들 화면 수정

devPaidion committed:
🔧[Chore]#46 - 애플 제공 로그인 버튼 사용

- 버튼의 액션 분리를 위해 SignInWithAppleButtonViewRepresentable 사용
- 버튼 높이 54로 고정

Velko716 commented on DanceMachine/DanceMachine/Common/Components/Login/CustomAppleSignInButton.swift:
```diff
@@ -0,0 +1,39 @@
+      }
+      .frame(height: 54)
+      .frame(maxWidth: .infinity)
+      .foregroundColor(.black)
```
.foregroundColor => .foregroundStyle로 변경해주세요.

devPaidion replied:
앗... 이런 실수를 부끄럽군요... 삭제할거라... 수정은 안 해도 될 것 같아요!

Velko716 commented on DanceMachine/DanceMachine/Common/Components/Login/CustomAppleSignInButton.swift:
이 뷰 삭제되는거죠?

devPaidion replied:
네!  삭제할게요...!

Velko716 reviewed (commented):
<img width="590" height="1278" alt="스크린샷, 2025-11-09 오전 12 12 13" src="https://github.com/user-attachments/assets/b5a15ab9-4d1e-4249-aa92-14e058f0ed22" />

이 부분 이미지도 수정해야 할 것 같습니다..! 
나머지는 문제 없는거 같아요!! 수정되면 어프로브 하겠습니다.

devPaidion committed:
🔥[Del]#46 - 커스텀 애플 로그인 버튼 파일 삭제

devPaidion committed:
💄[Design]#46 - 계정 설정 화면 Hi-fi 디자인 반영

jHoon99 reviewed (commented):
항상 고생 많으십니다! 
뷰 로딩이 기본 프로그레스인데 디자이너와 협의가 된걸까요??

devPaidion commented:
@jHoon99 
카단! 일단은 스피너 적용은 하지 않은 상태로 머지하고 추후에 반영하도록 하겠습니다!
연락 못 받으시는 것 같아서 일단 기록용으로 여기에 코멘트 남겨요!

devPaidion committed:
📦[Add]#46 - Lottie 라이브러리 추가 (lottie-spm)

- lottie-spm 로 라이브러리 추가
- https://github.com/LottieFiles/lottie-ios 의 README.md 확인

devPaidion committed:
✨[Feat]#46 - SpinnerPurple 로티 파일 생성

devPaidion committed:
✨[Feat]#46 - 로딩 스피너 컴포넌트 생성

devPaidion committed:
💄[Design]#46 - 알림 / 개인정보처리방침 / 이용약관 화면에 로딩스피너 적용
